### PR TITLE
Feat: integrate uv in the ci pipelines and the docs

### LINKED ISSUE #1020: integrate uv in the ci pipelines and the docs
integrate uv in the ci pipelines and the docs. it will make the ci faster and result in better docs

### PR DESCRIPTION
**Pull request by Mr-Sunglasses**
## Description

This PR fixes #1020

## Summary

This PR add the [uv](https://docs.astral.sh/uv/) package manager to CI pipeline. This will help in increasing the CI speed.

## PR Checklist

Please ensure that:

- [X] The PR contains a descriptive title
- [X] The PR contains a descriptive summary of the changes
- [X] You build and test your changes before submitting a PR.
- [X] You have added relevant documentation
- [X] You have added relevant tests. We prefer integration tests wherever possible

## Pre-Commit Instructions:

- [X] Ensure that you have run the [pre-commit hooks](https://github.com/sparckles/robyn#%EF%B8%8F-to-develop-locally) on your PR.

### CODE REVIEW

**Comment by sansyrox** [comment]
@Mr-Sunglasses , is this PR complete?

**Comment by Mr-Sunglasses** [comment]
> @Mr-Sunglasses , is this PR complete?

No this PR is still WIP

File: .github/workflows/codspeed.yml [hunk 1]
```diff
@@ -9,31 +9,38 @@ on:
   # performance analysis in order to generate initial data.
   workflow_dispatch:
 
+env:
+  UV_SYSTEM_PYTHON: 1
```
Comment (sansyrox) [inline .github/workflows/codspeed.yml:13]: one q - why is this needed

File: .github/workflows/lint-pr.yml [hunk 2]
```diff
@@ -4,20 +4,26 @@ on:
   pull_request:
     branches: [main]
 
+
+env:
+  UV_SYSTEM_PYTHON: 1
+
```
Comment (sansyrox) [inline .github/workflows/lint-pr.yml:10]: same q

**Review by sansyrox — COMMENTED** [review]
Hey @Mr-Sunglasses 👋 

Welcome back to the repo 😄  Just one question before we can merge.

File: .github/workflows/codspeed.yml (hunk 1 above)
Comment (Mr-Sunglasses) [reply .github/workflows/codspeed.yml:13]: ref. https://docs.astral.sh/uv/guides/integration/github/#using-uv-pip

```
If using the uv pip interface instead of the uv project interface, uv requires a virtual environment by default. To allow installing packages into the system environment, use the --system flag on all uv invocations or set the UV_SYSTEM_PYTHON variable.

The UV_SYSTEM_PYTHON variable can be defined in at different scopes.

Opt-in for the entire workflow by defining it at the top level:
```

File: .github/workflows/lint-pr.yml (hunk 2 above)
Comment (Mr-Sunglasses) [reply .github/workflows/lint-pr.yml:10]: ref. https://docs.astral.sh/uv/guides/integration/github/#using-uv-pip

```
If using the uv pip interface instead of the uv project interface, uv requires a virtual environment by default. To allow installing packages into the system environment, use the --system flag on all uv invocations or set the UV_SYSTEM_PYTHON variable.

The UV_SYSTEM_PYTHON variable can be defined in at different scopes.

Opt-in for the entire workflow by defining it at the top level:
```

**Review by sansyrox — APPROVED** [review]
LGTM! Thank you @Mr-Sunglasses ✨
